src/core/openthread.cpp

```diff
@@ -66,9 +66,6 @@ static otDEFINE_ALIGNED_VAR(sInstanceRaw, sizeof(otInstance), uint64_t);
 otInstance *sInstance = NULL;
 #endif
 
-void OT_CDECL operator delete(void *, size_t) throw() { }
-void OT_CDECL operator delete(void *) throw() { }
```

Comment:
I see myself arguing about something that there should be no discussion.
We use delete in otInstnace that is statically allocated. Further, otInstance does not define any specific destructor, thus there is no need to make any delete calls at all.
@nibanks please specify a case where Windows would need to call delete on a statically allocated memory without destructor. If there is a case, I will apply appropriate changes, otherwise, we should remove all delete calls and especially remote delete implementations from a file used across all platforms.